### ISSUE TITLE
spot 5 vegetation layer and temp directories

### ISSUE THREAD

**Issue by cnschulz** [mholling/nswtopo#10]
Gday, 

Not sure where my issue lies: Im using the 8.6GB spot5 vegetation layer and get faileures when trying to create a temp directory. Perhaps this is a permissions thing but I dont know where to set ruby permissions (or if there _are_ any)

The following error is from my 1.0 install. In 1.1 I just get "Error: No vegetation data file specified"

```
C:\Users\schulzc\Downloads\nswtopo-master\test>ruby nswtopo.rb
Calculating map bounds...
Map details:
  name: map
  size: 237mm x 162mm
  scale: 1:25000
  rotation: 0.0 degrees
  extent: 5.9km x 4.1km
Creating: nsw.vegetation-spot5, 1187x815 (1.0Mpx) @ 5.0 m/px
convert.exe: unable to open image `C:/Users/schulzc/Downloads/nswtopo-master/tes
t/d20150813-7572-fjkuq/nsw.vegetation-spot5-clut.png': No such file or directory
 @ error/blob.c/OpenBlob/2692.
convert.exe: unable to open file `C:/Users/schulzc/Downloads/nswtopo-master/test
/d20150813-7572-fjkuq/nsw.vegetation-spot5-clut.png' @ error/png.c/ReadPNGImage/
3912.
convert.exe: no images defined `C:/Users/schulzc/Downloads/nswtopo-master/test/d
20150813-7572-fjkuq/nsw.vegetation-spot5-mask.png' @ error/convert.c/ConvertImag
eCommand/3212.
convert.exe: unable to open image `C:/Users/schulzc/Downloads/nswtopo-master/tes
t/d20150813-7572-fjkuq/nsw.vegetation-spot5-mask.png': No such file or directory
 @ error/blob.c/OpenBlob/2692.
convert.exe: unable to open file `C:/Users/schulzc/Downloads/nswtopo-master/test
/d20150813-7572-fjkuq/nsw.vegetation-spot5-mask.png' @ error/png.c/ReadPNGImage/
3912.
convert.exe: no images defined `C:/Users/schulzc/Downloads/nswtopo-master/test/d
20150813-7572-fjkuq/nsw.vegetation-spot5.png' @ error/convert.c/ConvertImageComm
and/3212.
C:/Ruby22-x64/lib/ruby/2.2.0/fileutils.rb:1391:in `initialize': No such file or
directory @ rb_sysopen - C:/Users/schulzc/Downloads/nswtopo-master/test/d2015081
3-7572-fjkuq/nsw.vegetation-spot5.png (Errno::ENOENT)
        from C:/Ruby22-x64/lib/ruby/2.2.0/fileutils.rb:1391:in `open'
        from C:/Ruby22-x64/lib/ruby/2.2.0/fileutils.rb:1391:in `copy_file'
        from C:/Ruby22-x64/lib/ruby/2.2.0/fileutils.rb:485:in `copy_file'
        from C:/Ruby22-x64/lib/ruby/2.2.0/fileutils.rb:402:in `block in cp'
        from C:/Ruby22-x64/lib/ruby/2.2.0/fileutils.rb:1570:in `block in fu_each
_src_dest'
        from C:/Ruby22-x64/lib/ruby/2.2.0/fileutils.rb:1586:in `fu_each_src_dest
0'
        from C:/Ruby22-x64/lib/ruby/2.2.0/fileutils.rb:1568:in `fu_each_src_dest
'
        from C:/Ruby22-x64/lib/ruby/2.2.0/fileutils.rb:401:in `cp'
        from nswtopo.rb:1304:in `block in create'
        from nswtopo.rb:67:in `block in mktmppath'
        from C:/Ruby22-x64/lib/ruby/2.2.0/tmpdir.rb:88:in `mktmpdir'
        from nswtopo.rb:66:in `mktmppath'
        from nswtopo.rb:1303:in `create'
        from nswtopo.rb:2975:in `block in run'
        from nswtopo.rb:77:in `yield'
        from nswtopo.rb:77:in `block (2 levels) in recover'
        from nswtopo.rb:75:in `each'
        from nswtopo.rb:75:in `block in recover'
        from nswtopo.rb:2974:in `each'
        from nswtopo.rb:2974:in `each'
        from nswtopo.rb:2974:in `run'
        from nswtopo.rb:3145:in `<main>'
```

here is my config file

```
bounds: bounds.gpx
margin: 15

scale: 25000

rotation: 0

include:
#- nsw/vegetation-2008-v2
- nsw/vegetation-spot5
- declination
- relief
- nsw/topographic
- grid
- nsw/reserves
- nsw/holdings
- nsw/survey

nsw.vegetation-spot5:
  path: C:\Users\schulzc\Downloads\nswtopo-master\test\s5hgps_nsw_y20082012_bcvl0.tif
#  embed: false

#- plantation
#- nsw/reference-topo-current
#- nsw/aerial

#nsw.vegetation-2008-v2:
#  path: C:\Users\schulzc\Downloads\vegetation\Data\nswintext08\hdr.adf
#  colour:
#    woody: light green      # alternately specify a hex triplet, e.g. "#C2FFC2"
#    non-woody: white

#utm: true

grid:
  interval: 1000                 # interval between grid lines (1000 metres by default)
  width: 0.1                     # width in millimetres of the marked lines on the map
  #label-spacing: 5               # number of gridlines between successive labels
  label-spacing: false
  labels:
    font-size: 2.75              # font size (in mm) of the grid labels
    font-family: 'Arial Narrow'  # font family for the labels

declination:
  spacing: 1000           # perpendicular spacing of magnetic declination lines in metres
  width: 0.2              # width of the marked lines on the map, in millimetres
  stroke: red             # colour of magnetic declination lines (as a hex triplet or web colour)
  arrows: 50             # spacing in mm of directional arrows (set to 'false' for no arrows)


formats:
- png: 600
- map

#rasterise: C:\Users\schulzc\Downloads\nswtopo-master\test\phantomjs\bin\phantomjs.exe
#rasterise: C:\Users\schulzc\Downloads\batik-1.8
rasterise: C:\Program Files\Inkscape\inkscape.exe
```

**Comment by cnschulz** [comment]
Im not a ruby programmer but ive done some debugging to find out whats going on... its not the 8.6GB vegetation file... it seems its the temp file creation/permissions that are the problem. Ive proven that the gdla routines are working on the tif file. I do not see any temp dirs getting created when this script is run. Any ideas at all?

```
C:\nswtopo-master\test>gdalbuildvrt C:/nswtopo-master/test/d20150813-10044-11elv
ol/nsw.vegetation-spot5.vrt "C:/Users/schulzc/s5hgps_nsw_y20082012_bcvl0.tif"
0...10...20...30...40...50...60...70...80...90...100 - done.
ERROR 1: Failed to write .vrt file in FlushCache().

C:\nswtopo-master\test>gdalbuildvrt C:/nswtopo-master/test/d.vrt "C:/Users/schul
zc/s5hgps_nsw_y20082012_bcvl0.tif"
0...10...20...30...40...50...60...70...80...90...100 - done.
```

**Comment by cnschulz** [comment]
I have a "simple" workaround for this. I manually created a directory and put a file in it so the process would not delete it. I then changed the script and hardcoded the mktmpdir function create that directory. All the temp files are created in there now. Its not elegant and it still doesnt solve the issue of WHY I cant create temp directories...

**Comment by rjtedge** [comment]
Hi cnschulz

What part of the script did you change to get this working? Can you provide the code?

I'm also having this issue

**Comment by cnschulz** [comment]
Sure email me at cnschulz at gmail dot com. Although I think its fixed in v1.4 (Im now having a problem with relief layer.

Cheers

c.
